flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/QueryOperation.java

```diff
@@ -40,10 +41,12 @@ public interface QueryOperation extends Operation {
      * Returns a string that fully serializes this instance. The serialized string can be used for
      * storing the query in e.g. a {@link org.apache.flink.table.catalog.Catalog} as a view.
      *
+     * @param context can be used to customize the serialization to a SQL string
      * @return detailed string for persisting in a catalog
      * @see Operation#asSummaryString()
+     * @see EnvironmentSettings.Builder#withSerializationContext(SerializationContext)
      */
-    default String asSerializableString() {
+    default String asSerializableString(SerializationContext context) {
```

Comment:
This is a public evolving class. We need to keep the old asSerializableString around.

Reply:
Yes, I did it for `ResolvedExpression` but missed it here, sorry.